CMakeLists.txt

```diff
@@ -83,6 +83,13 @@ if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_LIST_DIR}")
   set(default_build_option_state ON)
 endif()
 
+foreach(arch IN LISTS CMAKE_CUDA_ARCHITECTURES)
+  if("${arch}" MATCHES "60")
+    add_compile_definitions(BUILD_PASCAL_CODE)
```

Comment:
Can't we just check `CUDA_ARCH`?

Reply:
`__CUDA_ARCH__` can't work for the host side code according to the [user guide](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#virtual-architecture-identification-macro).

Reply:
I still don't think this is quite the right place to do this. This will apply to all consumers of cuCo's cmake file, when we only want it applied to our tests/benchmarks.

CC @robertmaynard for the best way to do this.

Reply:
I'd also like to see this with a scoped and more descriptive name like `CUCO_HAS_INDEPENDENT_THREADS` or the inverse `CUCO_NO_INDEPENDENT_THREADS`.

Reply:
Added my recommendation for the CMake changes here https://github.com/NVIDIA/cuCollections/pull/107/files#r740458870